pytensor/xtensor/rewriting/shape.py

```diff
@@ -132,3 +139,32 @@ def local_squeeze_reshape(fgraph, node):
 
     new_out = xtensor_from_tensor(x_tensor_squeezed, dims=node.outputs[0].type.dims)
     return [new_out]
+
+
+@register_lower_xtensor
+@node_rewriter([ExpandDims])
+def local_expand_dims_reshape(fgraph, node):
+    """Rewrite ExpandDims to tensor.expand_dims and optionally broadcast_to or specify shape."""
+    x, size = node.inputs
+    out = node.outputs[0]
+    # Lower to tensor.expand_dims(x, axis=0)
+    from pytensor.tensor import expand_dims as tensor_expand_dims
+
+    expanded = tensor_expand_dims(tensor_from_xtensor(x), 0)
+    # Optionally broadcast to the correct shape if size is not 1
+    from pytensor.tensor import broadcast_to
+
+    # Ensure size is positive
+    expanded = Assert(msg="size must be positive")(expanded, gt(size, 0))
+    # If size is not 1, broadcast
+    try:
+        static_size = get_scalar_constant_value(size)
```

Comment:
You already inferred static size in make_node so you can use that info from `out.type.shape`